pkg/kube/http_utils.go

```diff
@@ -61,36 +61,35 @@ func (c *clientWithDefaultHeaders) Do(req *http.Request) (*http.Response, error)
 	return c.client.Do(req)
 }
 
+func GetAuthHeaderFromDockerConfig(dockerConfig *credentialprovider.DockerConfigJson) (string, error) {
+	// This is a simplified handler of a Docker config which only looks for the username:password
+	// of the first entry.
+	for _, entry := range dockerConfig.Auths {
+		auth := fmt.Sprintf("%s:%s", entry.Username, entry.Password)
+		authHeader := fmt.Sprintf("Basic %s", base64.StdEncoding.EncodeToString([]byte(auth)))
+		return authHeader, nil
+	}
+	return "", fmt.Errorf("The given config doesn't include an auth entry")
+}
+
 // getDataFromRegistrySecret retrieves the given key from the secret as a string
 func getDataFromRegistrySecret(key string, s *corev1.Secret) (string, error) {
-	dockerConfigJsonEncoded, ok := s.StringData[key]
+	dockerConfigJson, ok := s.Data[key]
 	if !ok {
-		authBytes, ok := s.Data[key]
+		authBytes, ok := s.StringData[key]
```

Comment:
So the only reason we check `StringData` here is because we used to be verifying a secret *before* the secret had been created, I assume? Otherwise `StringData` would not be a field at all. If we are now only ever verifying an existing secret (one that has been created already), we shouldn't need to use `StringData` at all - it's just a helper during secret creation.

Reply:
You checked all the call-sites for the function to be sure?